packages/marshaller/src/ddl2/activities/filter.ts

```diff
@@ -7,25 +7,37 @@ import { Activity, IActivityError, ReferencedFields } from "./activity";
 export class ColumnMapping extends PropertyExt {
     private _owner: Filter;
 
-    @publish(null, "set", "Filter Fields", function (this: ColumnMapping) { return this.sourceOutFields(); }, { optional: true })
+    @publish(null, "set", "Filter Fields", function (this: ColumnMapping) { return this.sourceOutFields(); }, {
+        optional: true,
+        disable: (w: ColumnMapping): boolean => !w._owner.source(),
+        validate: (w: ColumnMapping): boolean => w.sourceOutFields().indexOf(w.remoteField()) >= 0
+    })
     remoteField: publish<this, string>;
-    @publish(null, "set", "Local Fields", function (this: ColumnMapping) { return this.localFields(); }, { optional: true })
+    remoteField_exists: () => boolean;
+    remoteField_valid: () => boolean;
+    @publish(null, "set", "Local Fields", function (this: ColumnMapping) { return this.localFields(); }, {
+        optional: true,
+        disable: (w: ColumnMapping): boolean => !w._owner.source(),
+        validate: (w: ColumnMapping): boolean => w.localFields().indexOf(w.localField()) >= 0
+    })
     localField: publish<this, string>;
+    localField_exists: () => boolean;
+    localField_valid: () => boolean;
```

Comment:
I thought we were going to add all new publish param declarations in interfaces at the bottom of the widget files?

Reply:
Yes - this isn't a new one though - so I just changed the existing one.